### ISSUE TITLE
cypress.clock doesn't work with async await

### ISSUE THREAD

**Issue by shairez** [cypress-io/cypress#19620]
### Current behavior

I have an Angular app which fetches data via ajax.

After the data is being received there is an element showing in the template, and then after a timeout of 5000ms it disappears. 

I tried to uses `cy.clock` to speed things  up using `cy.tick(6000)` but it hangs because the code that should run after the ajax request returns  doesn't run.

When I remove the cy.clock everything works good (although delayed).			

### Desired behavior

I want to be able to use `cy.clock` and `cy.tick` with ajax calls in my E2E tests and speed things up.

I believe we might need to add `tickAsync` option to cypress.

### Test code to reproduce

I've uploaded a repository that simulates the issue (instructions are in the README) - 

https://github.com/shairez/cypress-clock-timeout-issue

### Cypress Version

9.2.0

### Other

I think it has to do with the fact it's async await related code.

I noticed sinon (which cy.clock is based on) has a `tickAsync` method, maybe that could be a good investigation route

**Comment by mjhenkes** [comment]
I believe you need to set `cy.clock` before calling `cy.visit`.
https://docs.cypress.io/api/commands/clock#Behavior

By making this change in your test case, the test passed.
```diff
describe('fetching products', () => {

  it('should fetch products and use fake timers', () => {

+  cy.clock();
    cy.visit('/');
-  cy.clock();
    cy.findByText('Fetch products').click();
    cy.findByText('fetched').should('exist');
    cy.findByText(/You pressed/).should('exist');
    cy.tick(6000);
    cy.findByText(/You pressed/).should('not.exist');

  })
})
```

**Comment by shairez** [comment]
Thanks @mjhenkes !
I verified it as well

This raises another issue (docs related) - 

For me, it wasn't clear from the documentation (I read it a few times before I opened this issue) that this is a viable solution to my problem because my `setTimeout` doesn't run automatically on page load, it runs after I emit an event (click the button).

Are you saying that as a requirement `cy.clock` should always be called before `cy.visit` ? 

if so, I would put it under "rules" and not "behavior"

If not, why does it work in this specific case then?

Thanks again!
